src/global/helpers/arrow-navigation.ts

```diff
@@ -0,0 +1,54 @@
+import getDocumentWritingMode from './get-document-writing-mode';
+
+/**
+ * Check if the key pressed is among those allowed for navigation.
+ * @param event The keyboard event to check.
+ */
+export function isArrowKeyPressed(event: KeyboardEvent): boolean {
+  return ['ArrowRight', 'ArrowLeft', 'ArrowUp', 'ArrowDown'].includes(event.key);
+}
+
+/**
+ * Checks the direction of the navigation event in the current writing mode.
+ * Returns the right function for calculate the next element's index,
+ * based on the of the current element's index and the size of the list to navigate.
+ * @param event The event to check.
+ * @returns A function to calculate the index of the element to move to:
+ * if it is a 'previous' event, returns a function to get the index of the previous element,
+ * or, if the current is the first in the list, the index of the last one;
+ * if it is a 'next' event, returns a function to get the index of the next element,
+ * or, if the current is the last in the list, the index of the first one.
+ */
+function getNextElementIndexFunction(
+  event: KeyboardEvent
+): (currentElementIndex: number, size: number) => number {
+  let prevKey: string, nextKey: string;
+  if (getDocumentWritingMode() === 'rtl') {
+    prevKey = 'ArrowRight';
+    nextKey = 'ArrowLeft';
+  } else {
+    prevKey = 'ArrowLeft';
+    nextKey = 'ArrowRight';
+  }
+
+  if (event.key === prevKey || event.key === 'ArrowUp') {
```

Comment:
Maybe to be a little bit shorter, math modulo (%) can also be used.

e.g. something like that
```
function go (offset) {
  currentElementIndex = (currentElementIndex + offset + size) % size
}
const next = () => go(1)
const previous = () => go(-1)
```